smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -574,4 +717,65 @@ contract BridgeBank is BankStorage,
 
         emit LogUnlock(recipient, token, amount);
     }
+
+    /**
+     * @notice Changes the denom of `_token` to `_denom`
+     * @dev Will set the denom both in this contract AND in the token itself
+     * @param _token Address of the BridgeToken
+     * @param _denom The Cosmos denom to be applied
+     * @return true if the operation succeeded
+     */
+    function setBridgeTokenDenom(
+      address _token, string memory _denom
+    ) public onlyOwner returns (bool) {
+      contractDenom[_token] = _denom;
+      return BridgeToken(_token).setDenom(_denom);
+    }
+
+    /**
+     * @notice Changes the denom of many tokens in a batch
+     * @dev Will set the denom both in this contract AND in each token
+     * @param _tokens List of address of BridgeTokens
+     * @param _denoms List of Cosmos denoms to be applied
+     * @return true if the operation succeeded
+     */
+    function batchSetBridgeTokenDenom(
+      address[] calldata _tokens, string[] calldata _denoms
+    ) external onlyOwner returns (bool) {
+      require(_tokens.length == _denoms.length, "INV_LEN");
+
+      for (uint256 i = 0; i < _tokens.length; i++) {
+        setBridgeTokenDenom(_tokens[i], _denoms[i]);
+      }
+
+      return true;
+    }
+
+    /**
+     * @notice Sets in this contract the denom of `_token`
+     * @dev Will fetch the denom from `_token` and register it in this contract
+     * @dev Anyone may call this function
+     * @param _token Address of the BridgeToken
+     * @return true if the operation succeeded
+     */
+    function forceSetBridgeTokenDenom(address _token) external returns (bool) {
```

Comment:
Let's enforce that the address is in the cosmos whitelist before allowing any setting to happen. So we can just pull in `onlyCosmosTokenWhitelist` modifier here to enforce this check.